### DISCUSSION TITLE
Voltage steps of 250mV

### DISCUSSION THREAD

**Discussion by thierryfischer** [oxplot/fpx#20] [Ideas]
Hi @oxplot , 
This is an amazing project that you have done.

I would like to know if that would be possible to configure the voltage with 0,25v steps as this is what is written in the datasheet of the IC. 

Best regards

**Comment by oxplot** [comment]
Thanks.

There are two things to keep in mind regarding the 50mV granularity mentioned in the datasheet:

1. You can specify that you **require** a certain voltage in 50mV increments, for example 8.45V. However, the USB-PD supply (your power adapter or power bank) must support this exact voltage for it to be available to the board. As far as consumer power adapters and power banks go, they only support a set of standard voltages, 5V, 9V, 12V, 15V and 20V (which are the ones available as selection for fpx board).
2. As you have noticed, fpx is programmed to only request the above standard voltages. If you require a custom voltage, you'll have to upload a custom firmware onto the board.

**Reply by thierryfischer** [reply]
Thanks for you clarification.
1. I got mixed up with the 0.2v steps of QC3 standard, indeed that is 50mV for USB-PD.

I had never take a look at the voltages of my USB-PD supply, only on my QC3 where steps are supported in available consumer power adapters.

2. Could you point out which files of the firmware have to be modified to support this.

**Reply by oxplot** [reply]
> Could you point out which files of the firmware have to be modified to support this.

The file of interest is https://github.com/oxplot/fpx/blob/master/firmware/main.c — note however that this isn't a one-line change and requires a good understanding of the registers on STUSB4500 chip. https://github.com/oxplot/fpx/blob/master/firmware/main.c#L311 is where the ATTiny programs the STUSB4500 chip.
